### PR TITLE
And/create channel tool

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description
This pull request allows to embed a channel in a unit course 

## Reviewers

- [ ] @jfavellar90 
- [ ] @felipemontoya

### CODE REVIEW

File: rocketc/static/html/rocketc.html [hunk 1]
```diff
@@ -1,16 +1,27 @@
 <div class="rocketc_block">
     <center>
+        {% if default_group_enable%}
 
-        {% if user_data.role == "instructor"%}
+            {% if "authToken" in response%}
+                <div class="embed-container">
+                    <iframe style="width:0;height:0;border:0; border:none;" src="http://localhost:3000/home?authToken={{response.authToken}}&userId={{response.userId}}" >IFrame is not supported</iframe>
+                    <iframe id="myframe" src="http://localhost:3000/group/{{user_data.default_group}}?layout=embedded" >IFrame is not supported</iframe>
```
Comment (jfavellar90) [inline rocketc/static/html/rocketc.html]: hardcoded urls

Comment (jfavellar90) [reply rocketc/static/html/rocketc.html]: or is a fallback?

File: rocketc/rocketc.py [hunk 2]
```diff
@@ -290,7 +309,29 @@ def _update_user(self, user_id, username, email):
...
             url_path = "users.update"
             data = {"userId": user_id, "data": {"email": email}}
-            response = self.request_rocket_chat("post", url_path, data)
+            response = self._request_rocket_chat("post", url_path, data)
             if response["success"]:
                 self.email = email
         self._set_avatar(username)
+
+    @XBlock.json_handler
+    def set_default_channel(self, data, suffix=""):
```
Comment (jfavellar90) [inline rocketc/rocketc.py]: Not an expert on xblocks, my question is when this method is called? what menas the decorator @XBlock.json_handler?

File: rocketc/static/js/src/rocketc.js [hunk 3]
```diff
@@ -1,22 +1,19 @@
...
 
-    function updateCount(result) {
-        $(".count", element).text(result.count);
-    }
+    $(function ($) {
+        /* Here's where you'd do things on page load. */
+    });
 
-    var handlerUrl = runtime.handlerUrl(element, "increment_count");
+    var setDefaultChannel = runtime.handlerUrl(element, "set_default_channel");
```
Comment (jfavellar90) [inline rocketc/static/js/src/rocketc.js]: This is something related to the method I ask you about above? the one I don't know where is called?

File: rocketc/rocketc.py [hunk 4]
```diff
@@ -290,7 +309,29 @@ def _update_user(self, user_id, username, email):
...
+        """
+        This method set the default variable for the channels
+        """
+        # pylint: disable=unused-argument
+        default_channel = data["channel"]
+        if default_channel != " " and default_channel != None:
```
Comment (jfavellar90) [inline rocketc/rocketc.py]: Here the second part of the conditional should be "and default_channel is not None"

File: rocketc/rocketc.py (hunk 2 above)
Comment (andrey-canon) [reply rocketc/rocketc.py]: This method is called by js when the person push a button and the decorator means that the method receives a json in data

File: rocketc/rocketc.py (hunk 4 above)
Comment (andrey-canon) [reply rocketc/rocketc.py]: ok I'll fix it

File: rocketc/static/js/src/rocketc.js (hunk 3 above)
Comment (andrey-canon) [reply rocketc/static/js/src/rocketc.js]: That is the way to get the method from python and use in js like a api rest

File: rocketc/static/html/rocketc.html (hunk 1 above)
Comment (andrey-canon) [reply rocketc/static/html/rocketc.html]: I fixed this in other branch

File: rocketc/static/js/src/rocketc.js [hunk 5]
```diff
@@ -1,22 +1,19 @@
...
-    $("p", element).click(function(eventObject) {
+    $("#button", element).click(function(eventObject) {
+        var channel = $("#name", element).val();
+        var data = {channel};
         $.ajax({
             type: "POST",
-            url: handlerUrl,
-            data: JSON.stringify({"hello": "world"}),
-            success: updateCount
+            url: setDefaultChannel,
```
Comment (jfavellar90) [inline rocketc/static/js/src/rocketc.js]: If we'are doing requests from frontend, we are going to use the public dns right?

Comment (andrey-canon) [reply rocketc/static/js/src/rocketc.js]: I have no idea how works

**Comment by jfavellar90** [comment]
lets merge this one

**Pull request merged**